kura/org.eclipse.kura.linux.position/src/main/java/org/eclipse/kura/linux/position/GpsDevice.java

```diff
@@ -314,11 +325,7 @@ public boolean doPollWork() {
 							c = in.read();
 						} catch (Exception e) {
 							s_logger.error("Exception in gps read - {}", e);
-							try {
-								Thread.sleep(1000);
-							} catch (InterruptedException e1) {
-								s_logger.warn("Interrupted - {}", e1);
```

Comment:
You are right. My goal was to simplify and make more readable the code. I'll encapsulate the sleep and catch in a private method.